Sources/NIO/IO.swift

```diff
@@ -14,15 +14,22 @@
 
 /// An `Error` for an IO operation.
 public struct IOError: Swift.Error {
+    @available(*, deprecated)
     public enum FailureDescription {
         case function(StaticString)
         case reason(String)
     }
-    /// The `errno` that was set for the operation.
-    public let errnoCode: Int32
 
     /// The actual reason (in an human-readable form) for this `IOError`.
-    public let reason: FailureDescription
+    private var failureDescription: String
+
+    @available(*, deprecated)
+    public var reason: FailureDescription {
+        return .reason(self.failureDescription)
+    }
+
+    /// The `errno` that was set for the operation.
+    public let errnoCode: Int32
```

Comment:
I know this is unrelated to this PR but the correct type should be `CInt` instead of `Int32`. So if you fancy, you could make that change (it’s API compatible on all supported platforms)